tests/unit/parts/test_lifecycle.py

```diff
@@ -198,23 +199,47 @@ def test_lifecycle_legacy_run_provider(cmd, snapcraft_yaml, new_dir, mocker):
         ("prime", "prime"),
     ],
 )
+@pytest.mark.parametrize("debug_shell", [None, "debug", "shell", "shell_after"])
 def test_lifecycle_run_command_step(
-    cmd, step, snapcraft_yaml, project_vars, new_dir, mocker
+    cmd, step, debug_shell, snapcraft_yaml, project_vars, new_dir, mocker
 ):
     project = Project.unmarshal(snapcraft_yaml(base="core22"))
     run_mock = mocker.patch("snapcraft.parts.PartsLifecycle.run")
     mocker.patch("snapcraft.meta.snap_yaml.write")
     pack_mock = mocker.patch("snapcraft.pack.pack_snap")
 
+    parsed_args = argparse.Namespace(
+        debug=False,
+        destructive_mode=True,
+        shell=False,
+        shell_after=False,
+        use_lxd=False,
+        parts=[],
+    )
+
+    if debug_shell:
+        setattr(parsed_args, debug_shell, True)
+
     parts_lifecycle._run_command(
-        cmd,
-        project=project,
-        parse_info={},
-        assets_dir=Path(),
-        parsed_args=argparse.Namespace(destructive_mode=True, use_lxd=False, parts=[]),
+        cmd, project=project, parse_info={}, assets_dir=Path(), parsed_args=parsed_args
     )
 
-    assert run_mock.mock_calls == [call(step)]
+    if debug_shell == "debug":
```

Comment:
you could avoid the repetition by storing the call arguments as a kwargs or by doing expected_call = call([...]) and doing a single check after the conditional

Reply:
Indeed, this can be written in a more compact way.